Sources/FoundationInternationalization/Calendar/Calendar_ICU.swift

```diff
@@ -1188,14 +1188,16 @@ internal final class _CalendarICU: _CalendarProtocol, @unchecked Sendable {
             if let value = components.weekdayOrdinal { ucal_set(ucalendar, UCAL_DAY_OF_WEEK_IN_MONTH, Int32(truncatingIfNeeded: value)) }
             // DateComponents month field is +1 from ICU
             if let value = components.month { ucal_set(ucalendar, UCAL_MONTH, Int32(truncatingIfNeeded: value - 1)) }
+
+            // The later the value is set via `ucal_set` the higher priority it takes when ICU resolves ambiguous components. For compatibility, always set `day of year` before `day (of month)`
```

Comment:
Alternatively we can also only set "day of year" if "day of month" isn't set, but I'm not sure how that'd work with other fields. ICU has its unique way of resolving combinations of fields, so I'd like to still set it always and let ICU handle it